src/software/backend/simulation/physics/physics_ball.cpp

```diff
@@ -19,21 +23,24 @@ PhysicsBall::PhysicsBall(std::shared_ptr<b2World> world, const Ball &ball)
     ball_body_def.bullet = true;
     ball_body            = world->CreateBody(&ball_body_def);
 
+    b2CircleShape ball_shape;
     ball_shape.m_radius = BALL_MAX_RADIUS_METERS;
 
+    b2FixtureDef ball_fixture_def;
     ball_fixture_def.shape = &ball_shape;
-
     // Calculate the density the fixture / ball must have in order for it to have the
     // desired mass. The density is uniform across the shape.
     float ball_area          = M_PI * ball_shape.m_radius * ball_shape.m_radius;
-    ball_fixture_def.density = BALL_MASS_KG / ball_area;
+    ball_fixture_def.density = mass_kg / ball_area;
     // These restitution and friction values are somewhat arbitrary. Because this is an
     // "ideal" simulation, we can approximate the ball as having perfectly elastic
     // collisions and no friction. Because we also do not generally depend on specific
     // behaviour when the ball collides with something, getting these values to perfectly
     // match reality isn't too important.
     ball_fixture_def.restitution = 1.0;
     ball_fixture_def.friction    = 0.0;
```

Comment:
nit: can we put these numbers in the header file

Reply:
done